### DISCUSSION TITLE
Support DEFAULT expression

### DISCUSSION THREAD

**Discussion by forrest-akin** [SeaQL/sea-query#99] [Q&A] [answered]
Column defaults in CREATE TABLE statements should, in addition to value literals, support expressions. For example, it's very common for a column of type `uuid` to default to either `gen_random_uuid()` or `uuid_generate_v4()`. Ideally, `ColumnSpec::Default(Value)` would contain a `SimpleExpr` instead of `Value` since `SimpleExpr` has a`Value` variant. This obviously would be a breaking change so perhaps instead consider adding `ColumnSpec::DefaultExpr(SimpleExpr)`.

**Comment by tyt2y3** [comment, marked as answer]
The [`extra()`](https://docs.rs/sea-query/0.12.8/sea_query/table/struct.ColumnDef.html#method.extra) method can be used to append expressions to the column definition.

I think default value and default expression are different things, but yes, if we have a `DefaultExpr` variant, we could probably make the `default` method to accept both `Value` and `SimpleExpr`. That say, we do not have symbols for all the database builtin functions yet (that'd be a nice-to-have), so perhaps for now `extra()` is your friend.

**Comment by forrest-akin** [comment]
Nice, will give that a shot 🙏

**Comment by tyt2y3** [comment]
@forrest-akin got success?

**Comment by forrest-akin** [comment]
Oops, forgot to check back. Yup, `.extra()` works for now for default expressions 👌
